src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,67 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double CONSTANT_CASE = 10;
+    private static final double MAX_SIDE_LENGTH = 100;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        Color color = colorSupplier.getRandomColor();
+        switch (random.nextInt(5)) {
+            case 0:
+                return createSquare(color);
+            case 1:
+                return createRectangle(color);
+            case 2:
+                return createRightTriangle(color);
+            case 3:
+                return createCircle(color);
+            case 4:
+                return createIsoscelesTrapezoid(color);
+            default:
+                throw new RuntimeException("Unexpected figure type");

```

Comment:
better to use Default method
```suggestion
                return getDefaultFigure();
```